library/Zend/Form/Element.php

```diff
@@ -83,6 +86,20 @@ public function init()
     }
 
     /**
+     * Remove the password before rendering if the form fails in order to avoid any security issue
+     *
+     * @param  FormInterface $form
+     * @return mixed
+     * @see    \Zend\Form\Element\Password::prepareElement
+     */
+    public function prepareElement(FormInterface $form)
+    {
+        if ('password' == $this->getAttribute('type')) {
+            $this->setValue('');
```

Comment:
It is always called at `$form->prepare()` and because `$form->prepare()` is needed before rendering a form this will work just fine.

I agree ideally you should alter the view helper, but because an `<input type="password" />` can be an instance of `Zend\Form\Element` *or* `Zend\Form\Element\Password` and thus can be rendered by `Zend\Form\View\Helper\FormInput` *or* `Zend\Form\View\Helper\FormPassword` it needs fixing on two separate places. I am opening a new issue about this, because it affects all form elements.